x/poolmanager/types/params.go

```diff
@@ -11,24 +11,61 @@ import (
 
 // Parameter store keys.
 var (
-	KeyPoolCreationFee = []byte("PoolCreationFee")
+	KeyPoolCreationFee                                = []byte("PoolCreationFee")
+	KeyDefaultTakerFee                                = []byte("DefaultTakerFee")
+	KeyOsmoTakerFeeDistribution                       = []byte("OsmoTakerFeeDistribution")
+	KeyNonOsmoTakerFeeDistribution                    = []byte("NonOsmoTakerFeeDistribution")
+	KeyAdminAddresses                                 = []byte("AdminAddresses")
+	KeyCommunityPoolDenomToSwapNonWhitelistedAssetsTo = []byte("CommunityPoolDenomToSwapNonWhitelistedAssetsTo")
+	KeyAuthorizedQuoteDenoms                          = []byte("AuthorizedQuoteDenoms")
 )
 
 // ParamTable for gamm module.
 func ParamKeyTable() paramtypes.KeyTable {
 	return paramtypes.NewKeyTable().RegisterParamSet(&Params{})
 }
 
-func NewParams(poolCreationFee sdk.Coins) Params {
+func NewParams(poolCreationFee sdk.Coins,
+	defaultTakerFee sdk.Dec,
+	osmoTakerFeeDistribution, nonOsmoTakerFeeDistribution TakerFeeDistributionPercentage,
+	adminAddresses, authorizedQuoteDenoms []string,
+	communityPoolDenomToSwapNonWhitelistedAssetsTo string) Params {
 	return Params{
 		PoolCreationFee: poolCreationFee,
+		TakerFeeParams: TakerFeeParams{
+			DefaultTakerFee:                                defaultTakerFee,
+			OsmoTakerFeeDistribution:                       osmoTakerFeeDistribution,
+			NonOsmoTakerFeeDistribution:                    nonOsmoTakerFeeDistribution,
+			AdminAddresses:                                 adminAddresses,
+			CommunityPoolDenomToSwapNonWhitelistedAssetsTo: communityPoolDenomToSwapNonWhitelistedAssetsTo,
+		},
+		AuthorizedQuoteDenoms: authorizedQuoteDenoms,
 	}
 }
 
 // DefaultParams are the default poolmanager module parameters.
 func DefaultParams() Params {
 	return Params{
 		PoolCreationFee: sdk.Coins{sdk.NewInt64Coin(appparams.BaseCoinUnit, 1000_000_000)}, // 1000 OSMO
+		TakerFeeParams: TakerFeeParams{
+			DefaultTakerFee: sdk.MustNewDecFromStr("0.0015"), // 0.15%
```

Comment:
I think we should be setting the precise param we want in the upgrade handler, and not have so many denoms in defaults